### PR TITLE
feat(proj-gen-gatsby): support for project generators using gatsby

### PR DESCRIPTION
**Pull request by JayaKrishnaNamburu**
- [x]  Support for a custom HTML file

### CODE REVIEW

**Comment by JayaKrishnaNamburu** [comment]
I feel the Gatsby community is more leaned towards the camelCase for css Modules. And after checking the current stage in which the generators are, I felt its good to switch to `camelCase` by default. So, in cases like preact we can configure them in their project generators when building.

**Comment by JayaKrishnaNamburu** [comment]
Currently, this creates projects powered by gatsby, and the custom Html can be controlled. Regarding `manifest` i have few ideas as we can consider them as framework-specific changes. Will make them in a separate PR

File: packages/teleport-project-generator-gatsby/README.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# teleport-project-generator-gatsby
+
+Project generator using React and React-Router, on top of a standard webpack config.
```
Comment (alexnm) [inline packages/teleport-project-generator-gatsby/README.md]: update description

File: packages/teleport-project-generator-gatsby/__tests__/end2end/index.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
+// @ts-ignore
```
Comment (alexnm) [inline packages/teleport-project-generator-gatsby/__tests__/end2end/index.ts]: you don't need the @ts-ignore for the json files anymore

File: packages/teleport-project-generator-gatsby/src/gatsby-mapping.json [hunk 3]
```diff
@@ -0,0 +1,18 @@
+{
+  "elements": {
+    "navlink": {
+      "elementType": "Link",
+      "dependency": {
+        "type": "library",
+        "path": "gatsby",
+        "version": "^2.15.36",
```
Comment (alexnm) [inline packages/teleport-project-generator-gatsby/src/gatsby-mapping.json]: no need to specify the version here if we have it in package.json

**Pull request merged**